### PR TITLE
Update React to 17 and update related packages

### PR DESCRIPTION
**Pull request by orzechdev**
It updates:
- React version from 16 to 17.
- `react-error-boundary` to newest
- `react-inlinesvg` to newest
- `@apollo/client` from 3.4.15 to 3.5.10
- `apollo-upload-client` to newest

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### Do you want to run more stable tests?
Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [x] giftCard
3. [x] category
4. [x] collection
5. [x] attribute
6. [x] productType
7. [x] shipping
8. [x] customer
9. [x] permissions
10. [x] menuNavigation
11. [x] pages
12. [x] sales
13. [x] vouchers
14. [x] homePage
15. [x] login
16. [x] orders
17. [x] products
18. [x] app
19. [x] plugins
20. [x] translations
21. [x] navigation

CONTAINERS=1

### CODE REVIEW

File: src/components/AppLayout/AppLayout.tsx [hunk 1]
```diff
@@ -152,21 +149,6 @@ const AppLayout: React.FC<AppLayoutProps> = ({
     isMenuActive(location.pathname, menuItem),
   )?.id;
 
-  const reloadWindow = () => {
-    window.location.reload();
-  };
-
-  const handleErrorBack = () => {
-    navigate("/", { replace: true });
```
Comment (andrzejewsky) [inline src/components/AppLayout/AppLayout.tsx:160]: don't we want this anymore?

File: src/graphql/client.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export const apolloClient = new ApolloClient({
       },
     } as TypedTypePolicies,
   }),
-  link,
+  link: (link as unknown) as ApolloLink,
```
Comment (andrzejewsky) [inline src/graphql/client.ts:38]: why does not fit? can you share the error?

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

File: src/graphql/client.ts (hunk 2 above)
Comment (orzechdev) [reply src/graphql/client.ts]: I downgraded types from 17.0.1 to 17.0.0, it looks like it works without type casting, so I removed it

File: src/components/AppLayout/AppLayout.tsx (hunk 1 above)
Comment (orzechdev) [reply src/components/AppLayout/AppLayout.tsx:160]: It's now moved to the `ErrorPage`

**Review by Droniu — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**